src/lib/grammar.js

```diff
@@ -38,13 +38,29 @@ var PSEUDOS = {
     ':visited':         ':vi'
 };
 
-var PSEUDOS_INVERTED = _.invert(PSEUDOS);
-var PSEUDO_REGEX = [];
-for (var pseudo in PSEUDOS) {
-    PSEUDO_REGEX.push(pseudo);
-    PSEUDO_REGEX.push(PSEUDOS[pseudo]);
+var PSEUDO_ELEMENTS = {
+    '::before':         '::b',
+    '::after':          '::a',
+    '::first-letter':   '::fl',
+    '::first-line':     '::fli'
+};
+
+var PSEUDO_CLASSES_INVERTED = _.invert(PSEUDO_CLASSES);
+var PSEUDO_CLASS_REGEX = [];
+for (var pseudo in PSEUDO_CLASSES) {
+    PSEUDO_CLASS_REGEX.push(pseudo);
+    PSEUDO_CLASS_REGEX.push(PSEUDO_CLASSES[pseudo]);
+}
+PSEUDO_CLASS_REGEX = '(?:' + PSEUDO_CLASS_REGEX.join('|') + ')(?![a-z])';
+
+var PSEUDO_ELEMENTS_INVERTED = _.invert(PSEUDO_ELEMENTS);
```

Comment:
All this code duplication bugs me, I think I'll take a stab at cleaning it up...